@coderabbitai Still not fixed, give me full test code so that it is fixed

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 5 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  scripts/automatedTasks.test.ts > automatedTasks entry point > parameter parsing and logging > parses process.argv correctly to extract taskType and taskPayload
AssertionError: expected "vi.fn()" to be called with arguments: [ '{"title":"Game","platform":"PC"}' ]

Number of calls: 0

 ❯ scripts/automatedTasks.test.ts:191:52
    189|
    190|             try {
    191|                 expect(vi.mocked(turnStringToObj)).toHaveBeenCalledWit…
       |                                                    ^
    192|             } finally {
    193|                 process.argv = originalArgv;

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯

 FAIL  scripts/automatedTasks.test.ts > automatedTasks entry point > parameter parsing and logging > logs task type, payload string, and parsed payload for debugging
AssertionError: expected "log" to be called with arguments: [ 'Parameters' ]

Number of calls: 0

 ❯ scripts/automatedTasks.test.ts:209:39
    207|             try {
    208|                 // Check that console.log was called with debugging in…
    209|                 expect(consoleLogSpy).toHaveBeenCalledWith('Parameters…
       |                                       ^
    210|                 expect(consoleLogSpy).toHaveBeenCalledWith('Task type …
    211|                 expect(consoleLogSpy).toHaveBeenCalledWith('Payload as…

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯


 FAIL  scripts/automatedTasks.test.ts > automatedTasks entry point > error handling > logs a friendly message when an unknown task type is provided
AssertionError: expected "log" to be called with arguments: [ StringContaining{…} ]

Number of calls: 0

 ❯ scripts/automatedTasks.test.ts:230:39
    228|
    229|             try {
    230|                 expect(consoleLogSpy).toHaveBeenCalledWith(
       |                                       ^
    231|                     expect.stringContaining('Bip bip - Unknown or unha…
    232|                 );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/5]⎯

 FAIL  scripts/automatedTasks.test.ts > automatedTasks entry point > database integration > opens the database once at module load time
AssertionError: expected "vi.fn()" to be called 1 times, but got 0 times
 ❯ scripts/automatedTasks.test.ts:262:45
    260|
    261|             // The module opens the database immediately on import
    262|             expect(vi.mocked(openDatabase)).toHaveBeenCalledTimes(1);
       |                                             ^
    263|         });
    264|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/5]⎯

 FAIL  scripts/automatedTasks.test.ts > automatedTasks entry point > database integration > passes the opened database instance to task handlers
AssertionError: expected "vi.fn()" to be called with arguments: [ {}, Any<Object> ]

Number of calls: 0

 ❯ scripts/automatedTasks.test.ts:277:54
    275|
    276|             try {
    277|                 expect(vi.mocked(addGameToDatabase)).toHaveBeenCalledW…
       |                                                      ^
    278|                     mockDb,
    279|                     expect.any(Object)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/5]⎯

 Test Files  1 failed | 115 passed (116)
      Tests  5 failed | 616 passed (621)
   Start at  21:25:18
   Duration  63.47s (transform 2.75s, setup 13.91s, import 20.95s, tests 12.94s, environment 107.04s)

--- a/scripts/automatedTasks.test.ts
+++ b/scripts/automatedTasks.test.ts
@@ -1,41 +1,147 @@
-import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
 import type Database from 'better-sqlite3';
 
-// Mock the database
-vi.mock('./common/db', () => ({
+const mocks = vi.hoisted(() => ({
     openDatabase: vi.fn(),
-}));
-
-// Mock turnStringToObj before importing automatedTasks
-vi.mock('./tasks/common/utils', () => ({
     turnStringToObj: vi.fn(),
+    tasks: {
+        addGameToDatabase: vi.fn(),
+        updateGameInDatabase: vi.fn(),
+        deleteGameFromDatabase: vi.fn(),
+        addBacklogToDatabase: vi.fn(),
+        deleteBacklogFromDatabase: vi.fn(),
+        cleanBacklog: vi.fn(),
+        addSerieToDatabase: vi.fn(),
+        manageSerieInDatabase: vi.fn(),
+        manageDlcsInDatabase: vi.fn(),
+        addTestToDatabase: vi.fn(),
+        updateTestInDatabase: vi.fn(),
+        deleteTestFromDatabase: vi.fn(),
+        updateTierLists: vi.fn(),
+        copyCovers: vi.fn(),
+        addCover: vi.fn(),
+    },
 }));
 
-// Mock all task imports
-vi.mock('./tasks', () => ({
-    addGameToDatabase: vi.fn(),
-    updateGameInDatabase: vi.fn(),
-    deleteGameFromDatabase: vi.fn(),
-    addBacklogToDatabase: vi.fn(),
-    deleteBacklogFromDatabase: vi.fn(),
-    cleanBacklog: vi.fn(),
-    addSerieToDatabase: vi.fn(),
-    manageSerieInDatabase: vi.fn(),
-    manageDlcsInDatabase: vi.fn(),
-    addTestToDatabase: vi.fn(),
-    updateTestInDatabase: vi.fn(),
-    deleteTestFromDatabase: vi.fn(),
-    updateTierLists: vi.fn(),
-    copyCovers: vi.fn(),
-    addCover: vi.fn(),
+vi.mock('./common/db', () => ({
+    openDatabase: mocks.openDatabase,
 }));
+
+vi.mock('./tasks/common/utils', () => ({
+    turnStringToObj: mocks.turnStringToObj,
+}));
+
+vi.mock('./tasks', () => mocks.tasks);
+
+type TaskHandlerName = keyof typeof mocks.tasks;
+
+type TaskCase = {
+    taskType: string;
+    payloadString: string;
+    expectedPayload: object;
+    handlerName: TaskHandlerName;
+    hasPayload?: boolean;
+};
+
+const taskCases: TaskCase[] = [
+    {
+        taskType: 'ADD_GAME',
+        payloadString: '{"title":"Test Game"}',
+        expectedPayload: { title: 'Test Game' },
+        handlerName: 'addGameToDatabase',
+    },
+    {
+        taskType: 'UPDATE_GAME',
+        payloadString: '{"id":1,"title":"Updated Game"}',
+        expectedPayload: { id: 1, title: 'Updated Game' },
+        handlerName: 'updateGameInDatabase',
+    },
+    {
+        taskType: 'DELETE_GAME',
+        payloadString: '{"id":1}',
+        expectedPayload: { id: 1 },
+        handlerName: 'deleteGameFromDatabase',
+    },
+    {
+        taskType: 'ADD_BACKLOG',
+        payloadString: '{"gameId":1}',
+        expectedPayload: { gameId: 1 },
+        handlerName: 'addBacklogToDatabase',
+    },
+    {
+        taskType: 'DELETE_BACKLOG',
+        payloadString: '{"id":1}',
+        expectedPayload: { id: 1 },
+        handlerName: 'deleteBacklogFromDatabase',
+    },
+    {
+        taskType: 'CLEAN_BACKLOG',
+        payloadString: '{}',
+        expectedPayload: {},
+        handlerName: 'cleanBacklog',
+        hasPayload: false,
+    },
+    {
+        taskType: 'ADD_SERIE',
+        payloadString: '{"name":"Test Serie"}',
+        expectedPayload: { name: 'Test Serie' },
+        handlerName: 'addSerieToDatabase',
+    },
+    {
+        taskType: 'MANAGE_SERIE',
+        payloadString: '{"id":1,"action":"update"}',
+        expectedPayload: { id: 1, action: 'update' },
+        handlerName: 'manageSerieInDatabase',
+    },
+    {
+        taskType: 'MANAGE_DLCS',
+        payloadString: '{"gameId":1,"dlcs":[]}',
+        expectedPayload: { gameId: 1, dlcs: [] },
+        handlerName: 'manageDlcsInDatabase',
+    },
+    {
+        taskType: 'ADD_TEST',
+        payloadString: '{"name":"Test"}',
+        expectedPayload: { name: 'Test' },
+        handlerName: 'addTestToDatabase',
+    },
+    {
+        taskType: 'UPDATE_TEST',
+        payloadString: '{"id":1,"name":"Updated"}',
+        expectedPayload: { id: 1, name: 'Updated' },
+        handlerName: 'updateTestInDatabase',
+    },
+    {
+        taskType: 'DELETE_TEST',
+        payloadString: '{"id":1}',
+        expectedPayload: { id: 1 },
+        handlerName: 'deleteTestFromDatabase',
+    },
+    {
+        taskType: 'UPDATE_TIER_LIST',
+        payloadString: '{"tier":"S","games":[1,2]}',
+        expectedPayload: { tier: 'S', games: [1, 2] },
+        handlerName: 'updateTierLists',
+    },
+    {
+        taskType: 'COPY_COVERS',
+        payloadString: '{"source":"src","dest":"dst"}',
+        expectedPayload: { source: 'src', dest: 'dst' },
+        handlerName: 'copyCovers',
+    },
+    {
+        taskType: 'ADD_COVER',
+        payloadString: '{"gameId":1,"coverUrl":"url"}',
+        expectedPayload: { gameId: 1, coverUrl: 'url' },
+        handlerName: 'addCover',
+    },
+];
 
 describe('automatedTasks entry point', () => {
     let mockDb: Database.Database;
     let consoleLogSpy: ReturnType<typeof vi.spyOn>;
 
     beforeEach(() => {
-        vi.clearAllMocks();
         mockDb = {} as Database.Database;
         consoleLogSpy = vi.spyOn(console, 'log').mockImplementation(() => {});
     });
@@ -44,244 +150,104 @@ describe('automatedTasks entry point', () => {
         consoleLogSpy.mockRestore();
     });
 
+    async function runAutomatedTask(
+        taskType: string,
+        payloadString: string,
+        parsedPayload: object
+    ): Promise<void> {
+        vi.resetModules();
+        vi.clearAllMocks();
+
+        mocks.openDatabase.mockReturnValue(mockDb);
+        mocks.turnStringToObj.mockReturnValue(parsedPayload);
+
+        const originalArgv = process.argv;
+        process.argv = ['node', 'automatedTasks.ts', taskType, payloadString];
+
+        try {
+            await import('./automatedTasks');
+        } finally {
+            process.argv = originalArgv;
+        }
+    }
+
     describe('task execution', () => {
-        const testCases = [
-            {
-                taskType: 'ADD_GAME',
-                payloadString: '{"title":"Test Game"}',
-                expectedPayload: { title: 'Test Game' },
-                handlerName: 'addGameToDatabase',
-            },
-            {
-                taskType: 'UPDATE_GAME',
-                payloadString: '{"id":1,"title":"Updated Game"}',
-                expectedPayload: { id: 1, title: 'Updated Game' },
-                handlerName: 'updateGameInDatabase',
-            },
-            {
-                taskType: 'DELETE_GAME',
-                payloadString: '{"id":1}',
-                expectedPayload: { id: 1 },
-                handlerName: 'deleteGameFromDatabase',
-            },
-            {
-                taskType: 'ADD_BACKLOG',
-                payloadString: '{"gameId":1}',
-                expectedPayload: { gameId: 1 },
-                handlerName: 'addBacklogToDatabase',
-            },
-            {
-                taskType: 'DELETE_BACKLOG',
-                payloadString: '{"id":1}',
-                expectedPayload: { id: 1 },
-                handlerName: 'deleteBacklogFromDatabase',
-            },
-            {
-                taskType: 'CLEAN_BACKLOG',
-                payloadString: '{}',
-                expectedPayload: {},
-                handlerName: 'cleanBacklog',
-            },
-            {
-                taskType: 'ADD_SERIE',
-                payloadString: '{"name":"Test Serie"}',
-                expectedPayload: { name: 'Test Serie' },
-                handlerName: 'addSerieToDatabase',
-            },
-            {
-                taskType: 'MANAGE_SERIE',
-                payloadString: '{"id":1,"action":"update"}',
-                expectedPayload: { id: 1, action: 'update' },
-                handlerName: 'manageSerieInDatabase',
-            },
-            {
-                taskType: 'MANAGE_DLCS',
-                payloadString: '{"gameId":1,"dlcs":[]}',
-                expectedPayload: { gameId: 1, dlcs: [] },
-                handlerName: 'manageDlcsInDatabase',
-            },
-            {
-                taskType: 'ADD_TEST',
-                payloadString: '{"name":"Test"}',
-                expectedPayload: { name: 'Test' },
-                handlerName: 'addTestToDatabase',
-            },
-            {
-                taskType: 'UPDATE_TEST',
-                payloadString: '{"id":1,"name":"Updated"}',
-                expectedPayload: { id: 1, name: 'Updated' },
-                handlerName: 'updateTestInDatabase',
-            },
-            {
-                taskType: 'DELETE_TEST',
-                payloadString: '{"id":1}',
-                expectedPayload: { id: 1 },
-                handlerName: 'deleteTestFromDatabase',
-            },
-            {
-                taskType: 'UPDATE_TIER_LIST',
-                payloadString: '{"tier":"S","games":[1,2]}',
-                expectedPayload: { tier: 'S', games: [1, 2] },
-                handlerName: 'updateTierLists',
-            },
-            {
-                taskType: 'COPY_COVERS',
-                payloadString: '{"source":"src","dest":"dst"}',
-                expectedPayload: { source: 'src', dest: 'dst' },
-                handlerName: 'copyCovers',
-            },
-            {
-                taskType: 'ADD_COVER',
-                payloadString: '{"gameId":1,"coverUrl":"url"}',
-                expectedPayload: { gameId: 1, coverUrl: 'url' },
-                handlerName: 'addCover',
-            },
-        ];
+        it.each(taskCases)(
+            'executes $taskType with the parsed payload',
+            async ({ taskType, payloadString, expectedPayload, handlerName, hasPayload = true }) => {
+                await runAutomatedTask(taskType, payloadString, expectedPayload);
 
-        it.each(testCases)(
-            'executes $taskType task with the parsed payload',
-            async ({ taskType, payloadString, expectedPayload, handlerName }) => {
-                // Reset modules to ensure clean state
-                vi.resetModules();
+                const handler = mocks.tasks[handlerName];
 
-                const { openDatabase } = await import('./common/db');
-                const { turnStringToObj } = await import('./tasks/common/utils');
-                const tasks = await import('./tasks');
-
-                vi.mocked(openDatabase).mockReturnValue(mockDb);
-                vi.mocked(turnStringToObj).mockReturnValue(expectedPayload);
-
-                // Simulate process.argv
-                const originalArgv = process.argv;
-                process.argv = ['node', 'script.ts', taskType, payloadString];
-
-                try {
-                    // Dynamically import to pick up mocked argv
-                    await import('./automatedTasks');
-
-                    // Verify the correct handler was called
-                    const handler = vi.mocked(tasks[handlerName as keyof typeof tasks]);
-
-                    // CLEAN_BACKLOG is called with only db (no payload)
-                    if (taskType === 'CLEAN_BACKLOG') {
-                        expect(handler).toHaveBeenCalledWith(mockDb);
-                    } else {
-                        expect(handler).toHaveBeenCalledWith(mockDb, expect.any(Object));
-                    }
-                } finally {
-                    process.argv = originalArgv;
+                if (hasPayload) {
+                    expect(handler).toHaveBeenCalledOnce();
+                    expect(handler).toHaveBeenCalledWith(mockDb, expectedPayload);
+                } else {
+                    expect(handler).toHaveBeenCalledOnce();
+                    expect(handler).toHaveBeenCalledWith(mockDb);
                 }
             }
         );
     });
 
     describe('parameter parsing and logging', () => {
-        it('parses process.argv correctly to extract taskType and taskPayload', async () => {
-            const { openDatabase } = await import('./common/db');
-            const { turnStringToObj } = await import('./tasks/common/utils');
+        it('parses process.argv to get the task type and payload', async () => {
+            const payloadString = '{"title":"Game","platform":"PC"}';
+            const parsedPayload = { title: 'Game', platform: 'PC' };
 
-            vi.mocked(openDatabase).mockReturnValue(mockDb);
-            const mockPayload = { title: 'Game', platform: 'PC' };
-            vi.mocked(turnStringToObj).mockReturnValue(mockPayload);
+            await runAutomatedTask('ADD_GAME', payloadString, parsedPayload);
 
-            const originalArgv = process.argv;
-            process.argv = ['node', 'script.ts', 'ADD_GAME', '{"title":"Game","platform":"PC"}'];
-
-            try {
-                expect(vi.mocked(turnStringToObj)).toHaveBeenCalledWith('{"title":"Game","platform":"PC"}');
-            } finally {
-                process.argv = originalArgv;
-            }
+            expect(mocks.turnStringToObj).toHaveBeenCalledOnce();
+            expect(mocks.turnStringToObj).toHaveBeenCalledWith(payloadString);
+            expect(mocks.tasks.addGameToDatabase).toHaveBeenCalledWith(mockDb, parsedPayload);
         });
 
-        it('logs task type, payload string, and parsed payload for debugging', async () => {
-            const { openDatabase } = await import('./common/db');
-            const { turnStringToObj } = await import('./tasks/common/utils');
+        it('logs the task type, payload string, and parsed payload', async () => {
+            const payloadString = '{"title":"Test"}';
+            const parsedPayload = { title: 'Test' };
 
-            vi.mocked(openDatabase).mockReturnValue(mockDb);
-            vi.mocked(turnStringToObj).mockReturnValue({ title: 'Test' });
+            await runAutomatedTask('ADD_GAME', payloadString, parsedPayload);
 
-            const originalArgv = process.argv;
-            process.argv = ['node', 'script.ts', 'ADD_GAME', '{"title":"Test"}'];
-
-            try {
-                // Check that console.log was called with debugging info
-                expect(consoleLogSpy).toHaveBeenCalledWith('Parameters');
-                expect(consoleLogSpy).toHaveBeenCalledWith('Task type :', 'ADD_GAME');
-                expect(consoleLogSpy).toHaveBeenCalledWith('Payload as string :', '{"title":"Test"}');
-            } finally {
-                process.argv = originalArgv;
-            }
+            expect(consoleLogSpy).toHaveBeenCalledWith('Parameters');
+            expect(consoleLogSpy).toHaveBeenCalledWith('Task type :', 'ADD_GAME');
+            expect(consoleLogSpy).toHaveBeenCalledWith('Payload as string :', payloadString);
+            expect(consoleLogSpy).toHaveBeenCalledWith('Payload as object :', parsedPayload);
         });
     });
 
     describe('error handling', () => {
-        it('logs a friendly message when an unknown task type is provided', async () => {
-            const { openDatabase } = await import('./common/db');
-            const { turnStringToObj } = await import('./tasks/common/utils');
+        it('logs a message for an unknown task type', async () => {
+            await runAutomatedTask('UNKNOWN_TASK', '{}', {});
 
-            vi.mocked(openDatabase).mockReturnValue(mockDb);
-            vi.mocked(turnStringToObj).mockReturnValue({});
-
-            const originalArgv = process.argv;
-            process.argv = ['node', 'script.ts', 'UNKNOWN_TASK', '{}'];
-
-            try {
-                expect(consoleLogSpy).toHaveBeenCalledWith(
-                    expect.stringContaining('Bip bip - Unknown or unhandled task')
-                );
-            } finally {
-                process.argv = originalArgv;
-            }
+            expect(consoleLogSpy).toHaveBeenCalledWith(
+                'Bip bip - Unknown or unhandled task: "UNKNOWN_TASK"'
+            );
         });
 
-        it('does not execute any task handler for an unknown task type', async () => {
-            const { openDatabase } = await import('./common/db');
-            const { turnStringToObj } = await import('./tasks/common/utils');
-            const { addGameToDatabase } = await import('./tasks');
+        it('does not execute a task handler for an unknown task type', async () => {
+            await runAutomatedTask('NONEXISTENT_TASK', '{}', {});
 
-            vi.mocked(openDatabase).mockReturnValue(mockDb);
-            vi.mocked(turnStringToObj).mockReturnValue({});
-
-            const originalArgv = process.argv;
-            process.argv = ['node', 'script.ts', 'NONEXISTENT_TASK', '{}'];
-
-            try {
-                expect(vi.mocked(addGameToDatabase)).not.toHaveBeenCalled();
-            } finally {
-                process.argv = originalArgv;
+            for (const handler of Object.values(mocks.tasks)) {
+                expect(handler).not.toHaveBeenCalled();
             }
         });
     });
 
     describe('database integration', () => {
-        it('opens the database once at module load time', async () => {
-            const { openDatabase } = await import('./common/db');
+        it('opens the database once when the entry point loads', async () => {
+            await runAutomatedTask('UNKNOWN_TASK', '{}', {});
 
-            // The module opens the database immediately on import
-            expect(vi.mocked(openDatabase)).toHaveBeenCalledTimes(1);
+            expect(mocks.openDatabase).toHaveBeenCalledOnce();
+            expect(mocks.openDatabase).toHaveBeenCalledWith();
         });
 
-        it('passes the opened database instance to task handlers', async () => {
-            const { openDatabase } = await import('./common/db');
-            const { turnStringToObj } = await import('./tasks/common/utils');
-            const { addGameToDatabase } = await import('./tasks');
+        it('passes the opened database instance to the selected handler', async () => {
+            const parsedPayload = { title: 'Test' };
 
-            vi.mocked(openDatabase).mockReturnValue(mockDb);
-            vi.mocked(turnStringToObj).mockReturnValue({ title: 'Test' });
+            await runAutomatedTask('ADD_GAME', '{"title":"Test"}', parsedPayload);
 
-            const originalArgv = process.argv;
-            process.argv = ['node', 'script.ts', 'ADD_GAME', '{"title":"Test"}'];
-
-            try {
-                expect(vi.mocked(addGameToDatabase)).toHaveBeenCalledWith(
-                    mockDb,
-                    expect.any(Object)
-                );
-            } finally {
-                process.argv = originalArgv;
-            }
+            expect(mocks.tasks.addGameToDatabase).toHaveBeenCalledOnce();
+            expect(mocks.tasks.addGameToDatabase).toHaveBeenCalledWith(mockDb, parsedPayload);
         });
     });
-
 });

